[3087] Add zoom, fit to screen and full screen on Deck

florianbarbin opened the pull request:
Bug: https://github.com/eclipse-sirius/sirius-web/issues/3087

# Pull request template

## General purpose
What is the main goal of this pull request?
- [ ] Bug fixes
- [x] New features
- [ ] Documentation
- [ ] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [ ] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [ ] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [ ] Have the relevant issues been added to the pull request?
- [ ] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [ ] Have the relevant issues been added to the same project and milestone as the pull request?
- [ ] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

## Review

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

florianbarbin committed:
[3087] Add zoom, fit to screen and full screen on Deck

Bug: https://github.com/eclipse-sirius/sirius-web/issues/3087
Signed-off-by: Florian Barbin <florian.barbin@obeo.fr>

pcdavid commented on packages/deck/frontend/sirius-components-deck/src/toolbar/DeckToolbar.tsx:
```diff
@@ -56,24 +65,54 @@ export const Toolbar = ({ editingContextId, representationId }: ToolbarProps) =>
+            size="small"
+            color="inherit"
+            aria-label="exit full screen mode"
+            title="Exit full screen mode"
```
Prefer MUI's `<Tooltip title="...">...</Tooltip>` instead of native tooltip.
It's what we use on [diagram](https://github.com/eclipse-sirius/sirius-web/blob/master/packages/diagrams/frontend/sirius-components-diagrams/src/renderer/palette/Palette.tsx#L466) and [portals](https://github.com/eclipse-sirius/sirius-web/pull/3040) for exemple so it would be more consistent.

florianbarbin replied:
done

pcdavid commented on packages/deck/frontend/sirius-components-deck/src/Deck.tsx:
```diff
@@ -32,32 +35,75 @@ export const Deck = ({
+    deckContainer: {
+      backgroundColor: theme.palette.background.default,
+    },
+  }));
```
Should be defined outside of the component.

florianbarbin replied:
done

florianbarbin committed:
[3087] Fix zoom issues

The ghost card or lane created by react-trello depends on an other component "react-smooth-dnd". This component create the "ghost" with a position "fixed" relative to first container. But when transformation is defined in the hierarchy (needed to implement the zoom feature), the position is relative to the first ancestor without transform property. That why we need to specify the ghost parent to avoid a delta between the ghost card / lane and the mouse.

Bug: https://github.com/eclipse-sirius/sirius-web/issues/3087
Signed-off-by: Florian Barbin <florian.barbin@obeo.fr>
Signed-off-by: Florian ROUËNÉ <florian.rouene@obeosoft.com>

frouene commented:
In full screen, the button fit to screen disappears

pcdavid commented:
> In full screen, the button fit to screen disappears

It's mentioned in the code comment that this is on purpose:

> //We disable the Fit to Screen but in Full screen mode because of issues to compute the parent container size.

frouene commented:
Failled test is not related to the PR
